6.bulk_Morphology_Elastic_Network/2.test_model/scripts/1.test_regression_multi_output.py

```diff
@@ -227,67 +210,65 @@
 # In[ ]:
 
 
-for data_split in data_dict:
-    print(data_split)
-    for i in data_dict[data_split]:
-        print(i)
-        data_x = data_dict[data_split]["data_x"]
-        data_y = data_dict[data_split]["data_y"]
-        col_names = data_dict[data_split]["col_names"]
-        metadata = data_dict[data_split]["metadata"]
-    if shuffle == "shuffled_baseline":
-        model = joblib.load(
-            f"../../1.train_models/{model_path}/{cytokine}_shuffled_baseline__all_nomic.joblib"
-        )
-    elif shuffle == "final":
-        model = joblib.load(
-            f"../../1.train_models/{model_path}/{cytokine}_final__all_nomic.joblib"
-        )
-    else:
-        print("Error")
-
-    # get the cytokine column of choice
-    y_selected = data_y[cytokine]
-
-    if shuffle == "shuffled_baseline":
-        for column in data_x:
-            np.random.shuffle(data_x[column].values)
-
-    # get predictions
-    predictions = model.predict(data_x)
-    for metric in metrics:
-        scores = cross_val_score(
-            model, data_x, y_selected, scoring=metric, cv=loo, n_jobs=-1
-        )
-        output_metric_scores[metric] = scores
-    r2 = r2_score(y_selected, predictions)
-    output_metric_scores["treatment"] = metadata[
-        "oneb_Metadata_Treatment_Dose_Inhibitor_Dose"
-    ]
-    output_metric_scores["well"] = metadata["Metadata_Well"]
-    df = pd.DataFrame.from_dict(output_metric_scores)
-    df["r2"] = r2
-    df["cytokine"] = cytokine
-    df["data_split"] = data_split
-    df["shuffle"] = shuffle
-    df["predicted_value"] = predictions
-    df["actual_value"] = y_selected
-    df["log10_neg_mean_absolute_error"] = -np.log10(-df["neg_mean_absolute_error"])
-    df["log10_neg_mean_squared_error"] = -np.log10(-df["neg_mean_squared_error"])
-    df["log10_explained_variance"] = -np.log10(df["explained_variance"])
-
-    # replace "[NSU]" with """
-    df["cytokine"] = df["cytokine"].replace("[ \[\]NSU]", "", regex=True)
-    df["cytokine"] = df["cytokine"].replace(" ", "_", regex=True)
-
-    # concat the dataframes
-    results_df = pd.concat([results_df, df], axis=0)
+data_x = test_data_x
+data_y = test_data_y
```

Comment:
this is a bit dangerous (without a deep copy) if you use any of the test data later in this notebook

Reply:
I learned what a deep copy is. I will implement in the future